hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -140,10 +157,99 @@ protected Path makeNewFilePath(String partitionPath, String fileName) {
    * Creates an empty marker file corresponding to storage writer path.
    *
    * @param partitionPath Partition path
+   * @param dataFileName data file for which inprogress marker creation is requested
+   * @param markerInstantTime - instantTime associated with the request
+   * returns true - inprogress marker successfully created,
+   *         false - inprogress marker was not created.
+   */
+  protected void createInProgressMarkerFile(String partitionPath, String dataFileName, String markerInstantTime) {
+    WriteMarkers writeMarkers = WriteMarkersFactory.get(config.getMarkersType(), hoodieTable, instantTime);
+    if (!writeMarkers.doesMarkerDirExist()) {
+      throw new HoodieIOException(String.format("Marker root directory absent : %s/%s (%s)",
+          partitionPath, dataFileName, markerInstantTime));
+    }
+    writeMarkers.create(partitionPath, dataFileName, getIOType());
+  }
+
+  protected boolean recoverWriteStatusIfAvailable(String partitionPath, String dataFileName,
+                                                  String markerInstantTime) {
+    WriteMarkers writeMarkers = WriteMarkersFactory.get(config.getMarkersType(), hoodieTable, instantTime);
+    if (config.isFailRetriesAfterFinalizeWrite()
+        && writeMarkers.markerExists(writeMarkers.getCompletionMarkerPath(StringUtils.EMPTY_STRING,
+        FINALIZE_WRITE_COMPLETED, markerInstantTime, IOType.CREATE))) {
+      throw new HoodieCorruptedDataException(" Failing retry attempt for instant " + instantTime
```

Comment:
is n't this actually preventing a re-attempt of the write, failing the stray task? i.e thus preventing corruption. Throwing "CorruptedData" exception makes sense if the table is deemed corrupted at this point, if not, then lets please rename the exception